JS/Lesson18/goksel/script.js

```diff
@@ -27,17 +36,27 @@ console.log('----------------------------');
 // 3. Write a function that takes an array of strings and concatenates them together adding spaces.
 // Input: ["Hello", "World", "!"]
 // Output: "Hello World !"
-let stringEx3 = ["Hello", "World", "!"];
+let stringEx3 = ["Hello", "World", "!", "I am", "web developer", "?"];
 
 function concatenateStrings(array) {
   let sentence = '';
   // TODO: if the next character is one of ?, '.', ',', etc. do not add a space before that character
   for (let i = 0; i < array.length; i++) {
-    sentence = sentence + ' ' + array[i];
+    const specialCharacter = ["?",".",",","!",";",":","'","-","_","="];
+    const nextArray = array[i+1];
+    if(specialCharacter.includes(nextArray)){
+      
+      sentence = sentence + " " + array[i];
+    }else {
+      sentence += array[i];
+    }
   }
   return sentence; 
 };
 
+// I think it shouldnt work like that but its working.  "if(!specialCharacter.includes(nextArray))" ı think this is how it should work correct but its not working like that.
```

Comment:
The issue is caused by the fact that you check if the next word is in the special characters array but you actually need to check the current character. See the comment above with the code example!